k8s/configmap/configmap.go

```
@@ -1,89 +1,79 @@
+	return &ConfigMapWrapper{
+		Client:  client.CoreV1().ConfigMaps(namespace),
+		Name:    configMapName,
+		DataKey: dataKey,
 	}
-	c.Client = client
+}
 
-	err = c.Client.Get(ctx, c.Client.Namespace, c.Name, &cm)
-	if err != nil {
-		log.Fatal(err)
-	}
-	c.CM = &cm
+// GetLocation implements github.com/Fresh-Tracks/bomb-squad/config.Configurator
+func (c *ConfigMapWrapper) GetLocation() string {
+	return c.DataKey
 }
 
-// ReadRawData pulls in value from the `data` key in the ConfigMap as-is
-func (c *ConfigMap) ReadRawData(ctx context.Context, key string) []byte {
-	cm := corev1.ConfigMap{}
-	err := c.Client.Get(ctx, c.Client.Namespace, c.Name, &cm)
+// Read implements github.com/Fresh-Tracks/bomb-squad/config.Configurator
+func (c *ConfigMapWrapper) Read() []byte {
+	dataKey := c.GetLocation()
+	cm, err := c.Client.Get(c.Name, v1.GetOptions{})
 	if err != nil {
-		log.Fatal(err)
-	}
-	c.CM = &cm
-
-	if res, ok := c.CM.Data[key]; !ok {
+		log.Printf("Failed to get ConfigMap in preparation for Configurator.Read(): %s\n", err)
 		return []byte{}
-	} else {
-		return []byte(res)
 	}
+
+	d := cm.Data[dataKey]
+
+	return []byte(d)
 }
 
-func (c *ConfigMap) Update(ctx context.Context, cfg promcfg.Config) error {
-	b, err := yaml.Marshal(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
+// Write implements github.com/Fresh-Tracks/bomb-squad/config.Configurator
+func (c *ConfigMapWrapper) Write(data []byte) error {
 
-	c.CM.Data[c.Key] = string(b)
-	err = c.UpdateWithRetries(5)
-	if err != nil {
-		log.Fatal(err)
-	}
+	dataKey := c.GetLocation()
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Retrieve the latest version of ConfigMap before attempting update
+		// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
 
-	return nil
-}
+		cm, err := c.Client.Get(c.Name, v1.GetOptions{})
+		if err != nil {
+			panic(fmt.Errorf("Failed to get latest version of ConfigMap: %v", err))
```

Why not just return the error since you return an error anyway?